packages/superset-ui-chart-controls/src/sections/advancedAnalytics.tsx

```diff
@@ -0,0 +1,130 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { t, RollingType, ComparisionType } from '@superset-ui/core';
+import { ControlPanelSectionConfig } from '../types';
+import { formatSelectOptions } from '../utils';
+
+export const advancedAnalytics: ControlPanelSectionConfig = {
+  label: t('Advanced analytics'),
+  tabOverride: 'data',
+  description: t(
+    'This section contains options ' +
+      'that allow for advanced analytical post processing ' +
+      'of query results',
+  ),
+  controlSetRows: [
+    [<h1 className="section-header">{t('Rolling window')}</h1>],
+    [
+      {
+        name: 'rolling_type',
+        config: {
+          type: 'SelectControl',
+          label: t('Rolling function'),
+          default: null,
+          choices: [[null, t('None')]].concat(formatSelectOptions(Object.values(RollingType))),
+          description: t(
+            'Defines a rolling window function to apply, works along ' +
+              'with the [Periods] text box',
+          ),
+        },
+      },
+    ],
+    [
+      {
+        name: 'rolling_periods',
+        config: {
+          type: 'TextControl',
+          label: t('Periods'),
+          isInt: true,
+          description: t(
+            'Defines the size of the rolling window function, ' +
+              'relative to the time granularity selected',
+          ),
+        },
+      },
+    ],
+    [
+      {
+        name: 'min_periods',
+        config: {
+          type: 'TextControl',
+          label: t('Min periods'),
+          isInt: true,
+          description: t(
+            'The minimum number of rolling periods required to show ' +
+              'a value. For instance if you do a cumulative sum on 7 days ' +
+              'you may want your "Min Period" to be 7, so that all data points ' +
+              'shown are the total of 7 periods. This will hide the "ramp up" ' +
+              'taking place over the first 7 periods',
+          ),
+        },
+      },
+    ],
+    [<h1 className="section-header">{t('Time comparison')}</h1>],
+    [
+      {
+        name: 'time_compare',
+        config: {
+          type: 'SelectControl',
+          multi: true,
+          freeForm: true,
+          label: t('Time shift'),
+          choices: formatSelectOptions([
+            '1 day ago',
+            '1 week ago',
+            '28 days ago',
+            '30 days ago',
+            '52 weeks ago',
+            '1 year ago',
+            '104 weeks ago',
+            '2 years ago',
+          ]),
+          description: t(
+            'Overlay one or more timeseries from a ' +
+              'relative time period. Expects relative time deltas ' +
+              'in natural language (example:  24 hours, 7 days, ' +
+              '52 weeks, 365 days). Free text is supported.',
+          ),
```

Comment:
Idea for future improvement: I think it would be great to support time grain based offsets. We could do this by checking if the value is parseable to an int and the time grain is defined, and if so, we could then create a time shift based on that. So for instance, if the time grain is "Week", and we would enter the offsets `1`, `2` and `4`, it would translate into `1 week ago`, `2 weeks ago` and `4 weeks ago`.

Reply:
I totally agree. BTW, I would like to merge this advanced analysis function into the Adhoc metric. Maybe we need some better UI design.